src/connectors/h1_collect.py

```diff
@@ -0,0 +1,316 @@
+"""HackerOne Data
+Collect HackerOne reports and payment transactions using an API key
+"""
+import os
+from datetime import datetime
+from urllib.error import HTTPError
+
+import fire
+import requests
+from runners.helpers import db, log
+from runners.helpers.dbconfig import ROLE as SA_ROLE
+
+from connectors.utils import yaml_dump
+
+PAGE_SIZE = 500
+
+CONNECTION_OPTIONS = [
+    {
+        'name': 'api_token',
+        'title': "API Token",
+        'prompt': "Available in your H1 Settings",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+    {
+        'name': 'api_identifier',
+        'title': "API Identifier",
+        'prompt': "Available in your H1 Settings",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+    {
+        'name': 'program_name',
+        'title': "Program Name",
+        'prompt': "Your handle in Program Settings",
+        'type': 'str',
+        'required': True,
+    },
+    {
+        'name': 'account_id',
+        'title': "Account ID",
+        'prompt': "https://api.hackerone.com/core-resources/#programs-get-your-programs",
+        'type': 'str',
+        'required': False,
+    },
+]
+
+LANDING_TABLE_COLUMNS_TRANSACTIONS = [
+    ('timestamp', 'TIMESTAMP_LTZ(9)'),
+    ('type', 'string'),
+    ('activity_date', 'TIMESTAMP_LTZ(9)'),
+    ('activity_description', 'string'),
+    ('bounty_award', 'int'),
+    ('bounty_fee', 'int'),
+    ('debit_or_credit_amount', 'int'),
+    ('balance', 'int'),
+    ('id', 'number'),
+    ('url', 'string'),
+]
+
+LANDING_TABLE_COLUMNS_REPORTS = [
+    ('timestamp', 'TIMESTAMP_LTZ(9)'),
+    ('id', 'number'),
+    ('type', 'string'),
+    ('title', 'string'),
+    ('state', 'string'),
+    ('created_at', 'TIMESTAMP_LTZ(9)'),
+    ('vulnerability_information', 'string'),
+    ('triaged_at', 'TIMESTAMP_LTZ(9)'),
+    ('closed_at', 'TIMESTAMP_LTZ(9)'),
+    ('first_program_activity_at', 'TIMESTAMP_LTZ(9)'),
+    ('bounty_awarded_at', 'TIMESTAMP_LTZ(9)'),
+    ('rating', 'string'),
+    ('name', 'string'),
+    ('description', 'string'),
+    ('external_id', 'string'),
+    ('asset_identifier', 'string'),
+    ('awarded_amount', 'int'),
+    ('awarded_bonus_amount', 'int'),
+]
+
+
+# Perform the API call
+def load_data(url: str, token: str, api_identifier: str, params: dict = {}) -> dict:
+    headers: dict = {"Accept": "application/json"}
+
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        resp = requests.get(
+            url, auth=(api_identifier, token), params=params, headers=headers
+        )
+        resp.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
```

Comment:
does the exception not contain the URL? even if not, seems we log it via `log.debug` above so this is vestigial. please remove.

Reply:
We copied the template from merkai_devices.py (lines80-86), would you like it rectified there too?

Reply:
yes, altho iirc we don't have any Markai devices to test changes against so not urgent